src/components/view-3d/interaction-handler.ts

```diff
@@ -0,0 +1,57 @@
+import { PointerEvent } from "react-three-fiber/canvas";
+
+// All the events listed here:
+// https://github.com/react-spring/react-three-fiber#events
+// It might be necessary to add more in the future.
+export enum InteractionAction {
```

Comment:
Can this library support touch events? I can't place sparks on iPad using either the code in Master nor with this branch. Now that we've gone this far, it would be super to add this option so we don't have to exclude touchscreen users.

Reply:
Pointer events touch devices (note that it's a new abstraction that simplifies mouse and touch events handling). There were other issues that I solved in the last commit. I had to remove onClick handler from this code though, but I think it's fine.